.github/workflows/ci.yml

```diff
@@ -246,12 +242,58 @@ jobs:
         with:
           java-version: 1.8
 
-      - name: Build PHAR and run the PHAR-specific tests with it
-        run: ant build-phar-and-run-phar-specific-tests
+      - name: Build PHAR
+        run: ant phar-snapshot
 
-      - uses: actions/upload-artifact@v2
-        if: ${{ matrix.php-version == 8.1 }}
+      - name: Upload PHAR
+        uses: actions/upload-artifact@v2
         with:
           name: phpunit-snapshot-phar
           path: ./build/artifacts/phpunit-snapshot.phar
           retention-days: 7
+
+  test-phar:
+    name: Test PHAR
+
+    needs:
+      - build-phar
+
+    runs-on: ubuntu-latest
+
+    env:
+      PHP_EXTENSIONS: none, dom, json, fileinfo, iconv, libxml, mbstring, pdo_sqlite, phar, soap, tokenizer, xml, xmlwriter
+      PHP_INI_VALUES: assert.exception=1, phar.readonly=0, zend.assertions=1
+
+    strategy:
+      fail-fast: false
+      matrix:
+        php-version:
+          - "7.4"
+          - "8.0"
+          - "8.1"
+          - "8.2"
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install PHP with extensions
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-version }}
+          coverage: none
+          extensions: ${{ env.PHP_EXTENSIONS }}
+          ini-values: ${{ env.PHP_INI_VALUES }}
+          tools: none
+
+      - name: Download PHAR
+        uses: actions/download-artifact@v2
+        with:
+          name: phpunit-snapshot-phar
+          path: ./build/artifacts/phpunit-snapshot.phar
+
+      - name: Make PHAR executable
```

Comment:
@sebastianbergmann 

You are right, see https://github.com/actions/upload-artifact#permission-loss!